### PR TITLE
fix(core/managed): properly diff/patch mutable infra groupings

### PR DESCRIPTION
**Pull request by erikmunson**
tl;dr I mistakenly thought that these filter services generated new grouping data and then pushed that out to consumers, but in reality the new data gets generated and then is used to diff + patch the original objects from the first time the grouping was created. Because I didn't implement this logic in previous PRs, changes to the `isManaged` and `managedResourceSummary` fields were not being picked up until a full reload.

Normally I would say this is a horrible kludge, but I suppose it's just the (gross) way of things while there's still some angular floating around since we need stable object references for those directives/components to behave correctly.

### CODE REVIEW

File: app/scripts/modules/core/src/cluster/filter/ClusterFilterService.ts [hunk 1]
```diff
@@ -473,6 +473,12 @@ export class ClusterFilterService {
         if (oldGroup.entityTags || newGroup.entityTags) {
           oldGroup.entityTags = newGroup.entityTags;
         }
+        if (oldGroup.hasOwnProperty('isManaged') || newGroup.hasOwnProperty('isManaged')) {
```
Comment (erikmunson) [inline app/scripts/modules/core/src/cluster/filter/ClusterFilterService.ts]: checking for only the `isManaged` field is sufficient, because the two fields travel together.
